### PR TITLE
Adding test for negative resources app creation

### PR DESCRIPTION
**Pull request by jimenez**
adding test for #3193

### CODE REVIEW

File: src/test/scala/mesosphere/marathon/api/v2/AppsResourceTest.scala [hunk 1]
```diff
@@ -60,6 +60,20 @@ class AppsResourceTest extends MarathonSpec with MarathonActorSupport with Match
     JsonTestHelper.assertThatJsonString(response.getEntity.asInstanceOf[String]).correspondsToJsonOf(expected)
   }
 
+  test("Create a new app fails with Validation errors for negative resources") {
+    Given("An app with negative resources")
+    val app = AppDefinition(id = PathId("/app"), cmd = Some("cmd"), versionInfo = OnlyVersion(Timestamp.zero), mem = -128, cpus = -1, instances = -1)
```
Comment (kolloch) [inline src/test/scala/mesosphere/marathon/api/v2/AppsResourceTest.scala]: Nit-pick: 

* not only the resources are negative, it would be great to test one thing at a time.
* it would be nice to break the line here because e.g. in github you cannot see the whole thing without scrolling.

File: src/test/scala/mesosphere/marathon/api/v2/AppsResourceTest.scala [hunk 2]
```diff
@@ -60,6 +60,20 @@ class AppsResourceTest extends MarathonSpec with MarathonActorSupport with Match
...
+    val group = Group(PathId("/"), Set(app))
+    val plan = DeploymentPlan(group, group)
+    val body = Json.stringify(Json.toJson(app)).getBytes("UTF-8")
+    groupManager.updateApp(any, any, any, any, any) returns Future.successful(plan)
+    groupManager.rootGroup() returns Future.successful(group)
+
+    Then("A constraint violation exception is thrown")
+    val response = appsResource.create(body, false, auth.request)
+    response.getStatus should be(422)
```
Comment (kolloch) [inline src/test/scala/mesosphere/marathon/api/v2/AppsResourceTest.scala]: Is it possible to make the validation more specific? E.g. test for the specific error reported?

**Comment by aquamatthias** [comment]
Thanks!
